docs/maintenance/release-management.md

```diff
@@ -146,3 +146,16 @@ only need to replace ``master`` with ``maint-x.y`` branches.
     Be aware that when the independent modules are released, they will be
     picked up immediately by new InvenioRDM installations. This should not be
     an issues since the releases MUST be backward compatible.
+
+## Trunk based Development Workflow
+
+[Trunk based Development](https://trunkbaseddevelopment.com/) happens in master, where we work with `v(latest stable + 1).0.0bX.devY` versions (i.e. `v13.0.0b0.dev0` now).
+Changes in master usually won't have an out-of-the-box backwards-compatibility assurance (things might break, but we'll fix them ASAP since we want to be able to develop reliably).
+
+Stable releases, like e.g. `v12.0.x` now, are being maintained under their equivalent `maint-v12.0.x` branch.
+
+- They will receive bug fixes following semver, i.e. `v12.0.1`, etc.
+- Occasionally, if we see features in master that the community agrees they would like to see in the stable branches, we can organize efforts to backport them using a minor version bump in the stable release (e.g. `v12.1.x`). These minor releases will have the following assurances:
+    - The new features **MUST** be disabled by default using config feature flags
+    - If an instance doesn't enable or use the new features, their instance **MUST** still work
+    - If an instance wants to enable/use the new features, there **MAY** be some migration/upgrade recipes required to be run (most of which will not require downtime for the application, e.g. adding new search mapping fields + reindexing records)
```

Comment:
These are clarifying/language elements. Documenting how we use Trunk-Based Development is super valuable, thanks!

```suggestion
InvenioRDM follows [Trunk-Based Development](https://trunkbaseddevelopment.com/) on the main branch (`master`) of all its modules. Changes in `master` usually won't have an out-of-the-box backwards-compatibility assurance. Things might break, but we'll fix them ASAP, since we want to be able to develop reliably.

Stable releases are being maintained under their equivalent `maint` branches. For example, `v12.0.x` is maintained on the `maint-v12.0.x` branch of `invenio-app-rdm`. Those releases receive bug fixes following SemVer, e.g., `v12.0.1`, by backporting (usually via cherry-picking) the fixes from the `master` branch into the `maint` branch.

Occasionally, if we see features in `master` that the community agrees should be in the stable branches, we can organize efforts to backport them using a minor version bump in the stable release (e.g., `v12.1.x`). To qualify for such minor version bumps, the changes adhere to the following assurances:
    - The new features **MUST** be disabled by default using config feature flags
    - If an instance doesn't enable or use the new features, it **MUST** still work
    - If an instance wants to enable or use the new features, there **MAY** be some migration/upgrade recipes required to be run 
```

The 

> (most of which will not require downtime for the application, e.g. adding new search mapping fields + reindexing records)

is a bit hard to say anything definitive about, so I would remove it to keep things open. 

Also, when we do make a minor version bump, what happens to the `maint` branch? Its name doesn't reflect its version anymore. Perhaps we should only create e.g., `maint-v12.x` in the first place? That'd be my vote. We have included the minor version in the branch naming in the past though.